File: java-module/src/main/java/com/bigdataboutique/spark/SparkSessionProvider.java
package com.bigdataboutique.spark;

import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.SparkSession;

public class SparkSessionProvider {

    public final static SparkSession spark = SparkSession.builder().appName("exercise").master("local[4]").getOrCreate();
    public final static JavaSparkContext sc = new JavaSparkContext(spark.sparkContext());

}

File: scala-module/src/main/scala/com/bigdataboutique/spark/SparkSessionProvider.scala
package com.bigdataboutique.spark

import org.apache.spark.SparkContext
import org.apache.spark.sql.SparkSession

trait SparkSessionProvider {
  lazy val spark: SparkSession = SparkSession.builder.appName("exercise").master("local[4]").getOrCreate
  lazy val sc: SparkContext = spark.sparkContext
}
